text-balancer.js

```diff
@@ -1,154 +1,412 @@
+     * @private
+     * @param {String} word - Word to check.
+     */
+    var isLowercaseWord = function (word) {
+        // If no word is passed, return early.
+        if (!word) return;
+
+        // Get first letter of word.
+        var firstLetter = word.charAt(0);
+
+        // Returns true if the first letter is lowercase.
+        return firstLetter === firstLetter.toLowerCase();
+    };
+
+    /**
+     * Get new width of element. This function is used to change the
+     * width so that the first word of the second line fits onto
+     * the first line.
+     * @private
+     * @param {NodeList} element - Element to change width.
+     * @param {Number} index - Index of the first word of the second line.
+     * @returns {Number} - New width of the element.
+     */
+    var getNewWidth = function (element, index) {
+        // If no element or index is passed, return early.
+        if (!element || !index) return;
+
+        var newWidth = 0;
+        var text = element.innerHTML;
+        var initialDisplay = element.style.display;
+
+        /**
+         * Get the full text of the first line plus the
+         * first word of the second line.
+         */
+        var widthText = text.split(' ').slice(0, index + 1).join(' ');
+
+        /**
+         * Add the text to the HTML of the element and reset
+         * its width.
+         */
+        element.innerHTML = widthText;
+        element.style.width = 'auto';
+        element.style.maxWidth = '';
+        // If it's block it will take up the full width and the function won't work.
+        element.style.display = 'inline-block';
+
+        // Get the new width.
+        newWidth = element.offsetWidth;
+
+        // Reset element back to its initial state.
+        element.innerHTML = text;
+        element.style.display = initialDisplay;
+
+        return newWidth;
+    };
+
+    /**
+     * Make the element as narrow as possible while maintaining
+     * its current height(number of lines). Binary search.
+     * @private
+     * @param {NodeList} element - The element to be balanced.
+     * @param {Number} originalHeight - The original height of the element.
+     * @param {Number} bottomRange - Bottom range of the (potential) width.
+     * @param {Number} topRange - Top range of the (potential) width.
+     */
+    var squeezeContainer = function (element, originalHeight, bottomRange, topRange) {
+        // Set a variable for the average of the top and bottom range.
+        var mid;
+
+        /**
+         * If the bottom range is equal to or larger than the top range,
+         * we've found the new width of our container.
+         */
+        if (bottomRange >= topRange) {
+            /**
+             * Set new width of the container, rounding it up and adding
+             * one extra pixel to avoid some edge cases.
+             */
+            element.style.maxWidth = Math.ceil(topRange + 1) + 'px';
+
+            if (!settings.splitOnLowercase) {
```

Comment:
That setting is passed in by default as true so it should be fine! I'll take a look at the other stuff soon.